x/consensus/keeper/keeper.go

```diff
@@ -139,3 +140,43 @@ func (k Keeper) paramCheck(ctx context.Context, consensusParams cmtproto.Consens
 
 	return &nextParams, nil
 }
+
+// BlockGas returns the maximum gas allowed in a block.
+func (k Keeper) BlockGas(ctx context.Context) (uint64, error) {
+	params, err := k.ParamsStore.Get(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	return uint64(params.Block.MaxGas), nil
+}
+
+// AppVersion returns the current application version.
+func (k Keeper) AppVersion(ctx context.Context) (uint64, error) {
+	params, err := k.ParamsStore.Get(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	return params.Version.App, nil
+}
+
+// ValidatorPubKeyTypes returns the list of public key types that are allowed to be used for validators.
+func (k Keeper) ValidatorPubKeyTypes(ctx context.Context) ([]string, error) {
+	params, err := k.ParamsStore.Get(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return params.Validator.PubKeyTypes, nil
```

Comment:
shouldn't we nil check those? when we were queried they were always nil-checked.